provider/resource_keycloak_realm.go

```diff
@@ -425,9 +452,17 @@ func setRealmData(data *schema.ResourceData, realm *keycloak.Realm) {
 		smtpSettings["from_display_name"] = realm.SmtpServer.FromDisplayName
 		smtpSettings["envelope_from"] = realm.SmtpServer.EnvelopeFrom
 		smtpSettings["ssl"] = realm.SmtpServer.Ssl
-		smtpSettings["user"] = realm.SmtpServer.User
-		smtpSettings["password"] = realm.SmtpServer.Password
-		data.Set("smtp_server", schema.NewSet(func(i interface{}) int { return 1 }, []interface{}{smtpSettings}))
+
+		if realm.SmtpServer.Auth {
+			auth := make(map[string]interface{})
+
+			auth["username"] = realm.SmtpServer.User
+			auth["password"] = realm.SmtpServer.Password
+
+			smtpSettings["auth"] = []interface{}{auth}
+		}
+
+		data.Set("smtp_server", []interface{}{smtpSettings})
```

Comment:
This line is why `TypeList` is easier to work with than `TypeString` in my opinion.

Reply:
I will be using this.